scientific/src/commonMain/kotlin/ScientificValue.kt

```diff
@@ -33,7 +33,7 @@ interface ScientificValue<Quantity : PhysicalQuantity, Unit : ScientificUnit<Qua
     val value: Number
     val unit: Unit
 
-    override fun compareTo(other: ScientificValue<Quantity, *>): Int = unit.toSIUnit(decimalValue).compareTo(other.unit.toSIUnit(other.decimalValue))
+    override fun compareTo(other: ScientificValue<Quantity, *>): Int = unit.toSIUnit(value.toDecimal()).compareTo(other.unit.toSIUnit(other.value.toDecimal()))
```

Comment:
`val decimalValue: Decimal get() = value.toDecimal()` Whats the point of this change. Decimal value is literally a shortcut for value.toDecimal()

Reply:
Because `DimensionlessScientificValue` overrides  `decimalValue` (so that  `1(Percent).value = 1` but `1(Percent).decimalValue = 0.01`)

As you said in this file you treat decimalValue as a shortcut to value.toDecimal() so calculations ware wrong for  DimensionlessScientificValues

Reply:
Moved the discussion to the relevant code